src/plugins/intel_gpu/src/kernel_selector/cl_kernels/include/batch_headers/sub_group_shuffle.cl

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common.cl"
+
+#ifdef cl_intel_subgroups
+#define _sub_group_shuffle(v, c) intel_sub_group_shuffle(v, c)
+#define _sub_group_shuffle_up(c, n, d) intel_sub_group_shuffle_up(c, n, d)
+#define _sub_group_shuffle_down(c, n, d) intel_sub_group_shuffle_down(c, n, d)
+#else
+
+#define DECLARE_SUB_GROUP_SHUFFLE1(type, cast_type)                                               \
+inline type _sub_group_shuffle(type v, uint c) __attribute__((overloadable)) {                    \
+    return AS_TYPE(type, sub_group_broadcast(AS_TYPE(cast_type, v), c));                          \
```

Comment:
Good catch, though as I can see in most of the cases shuffle uses the same index for all WI in a sub-group. I've added several comments for such kind of things, so we can update the impl later in case if we see real issues with current one.